pkg/utils/foreignCluster/getForeignCluster.go

```diff
@@ -48,3 +55,129 @@ func GetOlderForeignCluster(
+	// this address can be overwritten setting this environment variable
+	address := authServiceAddress
+	if address != "" {
+		return address, nil
+	}
+
+	// get the authentication service  (the namespace is automatically inferred by the namespaced client).
+	var svc corev1.Service
+	ref := types.NamespacedName{Name: liqoconst.AuthServiceName, Namespace: namespace}
+	if err := namespacedClient.Get(ctx, ref, &svc); err != nil {
+		klog.Error(err)
+		return "", err
+	}
+
+	// if the service is exposed as LoadBalancer
+	if svc.Spec.Type == corev1.ServiceTypeLoadBalancer {
+		// get the IP from the LoadBalancer service
+		if len(svc.Status.LoadBalancer.Ingress) == 0 {
+			// the service has no external IPs
+			err := goerrors.New("no valid external IP for LoadBalancer Service")
+			klog.Error(err)
+			return "", err
+		}
+		lbIngress := svc.Status.LoadBalancer.Ingress[0]
+		// return the external service IP
+		if hostname := lbIngress.Hostname; hostname != "" {
+			return hostname, nil
+		} else if ip := lbIngress.IP; ip != "" {
+			return ip, nil
+		} else {
+			// the service has no external IPs
+			err := goerrors.New("no valid external IP for LoadBalancer Service")
+			klog.Error(err)
+			return "", err
+		}
+	}
+
+	// only physical nodes
+	//
+	// we need to get an address from a physical node, if we have established peerings in the past with other clusters,
+	// we may have some virtual nodes in our cluster. Since their IPs will not be reachable from other clusters, we cannot use them
+	// as address for a local NodePort Service
+	req, err := labels.NewRequirement(liqoconst.TypeLabel, selection.NotIn, []string{liqoconst.TypeNode})
+	utilruntime.Must(err)
+
+	// get the IP from the Nodes, to be used with NodePort services
+	nodes := corev1.NodeList{}
+	if err := clusterWideClient.List(ctx, &nodes, client.MatchingLabelsSelector{Selector: labels.NewSelector().Add(*req)}); err != nil {
+		klog.Error(err)
+		return "", err
+	}
+
+	if len(nodes.Items) == 0 {
+		// there are no node is the cluster, we cannot get the address on any of them
+		err = kerrors.NewNotFound(corev1.Resource("nodes"), "")
+		klog.Error(err)
+		return "", err
+	}
+
+	node := nodes.Items[0]
+	return discovery.GetAddress(&node)
+
+	// when an error occurs, it means that we was not able to get an address in any of the previous cases:
+	// 1. no overwrite variable is set
+	// 2. the service is not of type LoadBalancer
+	// 3. there are no nodes in the cluster to get the IP for a NodePort service
+}
+
+// getAuthPort the external port where the Authentication Service is reachable from the external world.
+func getAuthPort(ctx context.Context, cl client.Client, authServicePort, authServiceNamespace string) (string, error) {
+	// this port can be overwritten setting this environment variable
+	port := authServicePort
+	if port != "" {
+		return port, nil
+	}
```

Comment:
```suggestion
	// this port can be overwritten setting this environment variable
	if authServicePort != "" {
		return authServicePort, nil
	}
```

Should also update the comment